### PR TITLE
Lesson 6

### PR DESCRIPTION
**Pull request by Irina-pi**

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -1,49 +1,187 @@
...
-console.log(test);
-
-/*task 8*/
-let arr = [2, 3, 5, 8];
-let result = 1;
-
-for (let i = 0; i < arr.length; i++) {
-    result *= arr[i];
+//task 19.5
+const palindrome = (word) => word.charAt(0) === word.charAt(word.length - 1);
```
Comment (5tronciy) [inline main.js]: What about the words with more than 3 chars?

File: main.js [hunk 2]
```diff
@@ -1,49 +1,187 @@
...
+
+function searchByWord(word) {
+  const filterHotels = hotels.filter((itemObject) => {
+    return (
+      itemObject.name === word ||
+      itemObject.city === word ||
+      itemObject.country === word
+    );
+  });
+  return Object.values(filterHotels);
```
Comment (5tronciy) [inline main.js]: Check the format of the return with the task.

File: main.js [hunk 3]
```diff
@@ -1,49 +1,187 @@
...
-        console.log(arr9[i]);
-    }
-  }
-  return array;
-};
+//task 19.7
+function searchCity(word) {
+  const indexFilter = hotels.findIndex((item) => item.country === word);
+  return `${word}: ${hotels[indexFilter].city}`;
+}
```
Comment (5tronciy) [inline main.js]: It doesn't assign the task requirements.

**Review by 5tronciy — CHANGES_REQUESTED** [review]
Fix the issues, but first clean the repo of node_modules.

File: main.js (hunk 1 above)
Comment (Irina-pi) [reply main.js]: I maked this task correctly just now.

File: main.js (hunk 2 above)
Comment (Irina-pi) [reply main.js]: ok, I made this function by task.

File: src/main.js [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+  const result = hotels.reduce((acc, hotel) => {
+    if (hotel.name === word || hotel.city === word || hotel.country === word) {
+      acc.push(`${hotel.country}, ${hotel.city}, ${hotel.name}`);
+    }
+    return acc;
+  }, []);
+  if (!result.length) {
+    return 'Search didn`t get results';
+  }
+  return result.join('; ');
```
Comment (5tronciy) [inline src/main.js:178]:
```suggestion
  return result;
```
It's cleaner that way and assigns the task.

**Review by 5tronciy — APPROVED** [review]
All the mandatory requirements are met. Approve.

File: src/main.js (hunk 4 above)
Comment (Irina-pi) [reply src/main.js:178]: ok, thanks!

**Pull request merged**